### ISSUE TITLE
Bad perf on watch_children for wine processes caused by lsof

### ISSUE THREAD

**Issue by Xodetaetl** [lutris/lutris#182]
So I had this high cpu usage with Wine games after reducing the heartbeat to 100ms.
I did some profiling with the line_profiler module and found that [this line](https://github.com/lutris/lutris/blob/next/lutris/util/system.py#L184) takes 110ms of processing time with my cpu, it's 90% of `thread.watch_children`'s. 
(Edit: the other 10% are from reading the yaml files to get the wine version and custom exe)

It would be nice if we could find an alternative to using lsof. I really want to lower the heartbeat without impacting performance. 100ms is the maximum delay before being noticeable and annoying.

**Comment by strycore** [comment]
I tried to find alternatives to lsof, actually I tried that before writing the lsof solution, I asked on the Wine mailing list (no answer) and tried to see what I could get from the wineserver without any success.
I understand that this solution is very hackish and can have a huge impact on perfs if overused but it's currently the only thing available to monitor wine processes.
Now, I know that having the smallest heartbeat delay possible will improve the overall responsiveness but in this particular case, we can't lower this too much (I didn't notice any issues using a 1 sec delay).

There should be a way to keep the heartbeat delay low while keeping it high enough on wine games. lsof could have it's own heartbeat, when called it could return a timestamp telling the main monitoring thread whhen to run next time.

That said, if you find *anything* that could replace lsof. I'm all for it!
